src/test/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTripTest.java

```diff
@@ -730,6 +732,114 @@ void canMakeOTPRequestAndResolveNoLongerPossibleTrip() throws Exception {
         );
     }
 
+    /**
+     * Utility method to set delays on a transit itinerary.
+     */
+    void addTransitLegDelay(Itinerary itinerary, int departureDelay, int arrivalDelay, boolean realTime) {
+        Leg walkLeg = itinerary.legs.get(0);
+        Leg transitLeg = itinerary.legs.get(1);
+        Leg finalLeg = itinerary.legs.get(2);
+
+        walkLeg.startTime = Date.from(walkLeg.startTime.toInstant().plusSeconds(departureDelay));
+        walkLeg.endTime = Date.from(walkLeg.endTime.toInstant().plusSeconds(departureDelay));
+
+        transitLeg.realTime = realTime;
+        transitLeg.departureDelay += departureDelay;
+        transitLeg.startTime = Date.from(transitLeg.startTime.toInstant().plusSeconds(departureDelay));
+        transitLeg.arrivalDelay += arrivalDelay;
+        transitLeg.endTime = Date.from(transitLeg.endTime.toInstant().plusSeconds(arrivalDelay));
+
+        finalLeg.startTime = Date.from(finalLeg.startTime.toInstant().plusSeconds(arrivalDelay));
+        finalLeg.endTime = Date.from(finalLeg.endTime.toInstant().plusSeconds(arrivalDelay));
+
+        itinerary.startTime = Date.from(itinerary.startTime.toInstant().plusSeconds(departureDelay));
+        itinerary.endTime = Date.from(itinerary.endTime.toInstant().plusSeconds(arrivalDelay));
+    }
+
+    /**
+     * A trip delay notification should be sent when saving a trip that has delays,
+     * and it is possible to find a matching trip without delays with the trip time in the OTP query params.
+     */
+    @Test
+    void canSendDelayNotificationOnSavingDelayedTrip() throws Exception {
+        OtpResponse mockWeekdayResponse = mockOtpPlanResponse();
```

Comment:
Should I combine this test with `testDelayNotifications` (which is focused on the internals)?